backend/setup_env.py

```diff
@@ -0,0 +1,174 @@
+
+    backend_env["AUTH_CSE_CLIENT_ID"] = auth_cse_client_id
+    backend_env["AUTH_CSE_CLIENT_SECRET"] = auth_cse_client_secret
+    backend_env["AUTH_REDIRECT_BASE_URI"] = auth_cse_redirect_base_uri
+
+    backend_env["SESSIONSDB_USERNAME"] = sessionsdb_username
+    backend_env["SESSIONSDB_PASSWORD"] = sessionsdb_pass
+    backend_env["SESSIONSDB_SERVICE_HOSTNAME"] = sessionsdb_hostname
+
+    # redis - sessionsdb
+    base_redis_args = f"--user {sessionsdb_username} on allcommands allkeys allchannels >{sessionsdb_pass}"
+    additional_redis_args = env.get_variable("ADDITIONAL_REDIS_ARGS", "")
+    redis_args = f"{base_redis_args} {additional_redis_args}" if additional_redis_args else base_redis_args
+    sessionsdb_env["ADDITIONAL_REDIS_ARGS"] = additional_redis_args
+    sessionsdb_env["REDIS_ARGS"] = redis_args
+
+
+    # Misc - backend
+    python_version = env.get_variable("PYTHON_VERSION", "python3")
+    backend_env["PYTHON_VERSION"] = python_version
+
+    if env.in_production:
+        backend_env["FORWARDED_ALLOWED_IPS"] = "*"
+
+    # mongodb - backend + mongodb
+    mongo_username = env.get_variable('MONGODB_USERNAME', "name")
+    mongo_pass = env.get_variable('MONGODB_PASSWORD', "pass123")
+    mongo_hostname = env.get_variable("MONGODB_SERVICE_HOSTNAME", "mongodb")
+
+    backend_env["MONGODB_USERNAME"] = mongo_username
+    backend_env["MONGODB_PASSWORD"] = mongo_pass
+    backend_env["MONGODB_SERVICE_HOSTNAME"] = mongo_hostname
+
+    mongo_env["MONGO_INITDB_ROOT_USERNAME"] = mongo_username
+    mongo_env["MONGO_INITDB_ROOT_PASSWORD"] = mongo_pass
+
+
+
+    # frontend
+    vite_backend_base_url = env.get_variable("VITE_BACKEND_API_BASE_URL", "http://localhost:8000/")
+    vite_env = env.get_variable("VITE_ENV", "dev")
+    
+    frontend_env["VITE_BACKEND_API_BASE_URL"] = vite_backend_base_url
+    frontend_env["VITE_ENV"] = vite_env
+
+    # Finished reading - save all changes
+    write_env_file(backend_env, BACKEND_ENV)
+    write_env_file(frontend_env, FRONTEND_ENV)
+    write_env_file(mongo_env, MONGO_ENV)
+    write_env_file(sessionsdb_env, SESSIONSDB_ENV)
+    print("Finished writing all env files. Exiting successfully")
+
+class EnvReader():
+
+    def __init__(self, *, env_files: list[Path]) -> None:
+        cli_args = vars(parse_cli_args())
+
+        self.in_production = bool(cli_args.get("prod"))
+        self.cli_args: dict[str, str] = {
+            k: str(v) for k, v in cli_args.items() if v is not None
+        }
+
+        print("in prod", self.in_production)
+        self.preexisting_env: dict[str, str] = {}
+
+        for env_file in env_files:
+            parsed_env = {
+                key: value
+                for key, value in dotenv.dotenv_values(env_file).items()
+                if value is not None
+            }
+            self.preexisting_env.update(parsed_env)
+
+    def get_variable(self, name: str, default: Optional[str] = None) -> str:
```

Comment:
i think the `default` arg is unused here. Did you mean to use it as a fallback if no preexisting_env?

Reply:
Ahh yea, there was some changes when migrating this from `prompt_variable`. I've made it so that `default` is also now used in `get_variable`